### DISCUSSION TITLE
Appending the array of strings to a yaml list

### DISCUSSION THREAD

**Discussion by paradajz** [TomWright/dasel#221] [Q&A] [answered]
I have an array stored in bash variable containing data, eg:

```
line1
line2
line3
```

Now I want to append those lines to a single subkey, like so:

```
---
list:
 - line1
 - line2
 - line3
```

Can't figure out how to do that. The closest I got was this:

```
#!/bin/bash

filename=test.yaml
echo "---" > $filename
array=(line1 line2 line3)

for ((i=0;i<${#array[*]};i++))
do
    {
        echo "" |
        dasel put string -p yaml "list.[$i]" "${array[$i]}"
    } >> $filename
done
```

This produces the following:

```
---
list:
- line1
list:
- line2
list:
- line3

```

**Comment by TomWright** [comment, marked as answer]
The dasel command you're looking for is:
```bash
$ dasel put string -f test.yaml -o - "list.[]" "line4"
list:
- line1
- line2
- line3
- line4
```

Note that to append we use the `[]` selector rather than `[$i]`. Using `[$i]` you will be overwriting the value at that index.

```bash
$ dasel put string -f test.yaml -o - "list.[1]" "line4"
list:
- line1
- line4
- line3
```

Here's an example `run.sh` that will append the bash array items to `test.yaml` and write the results to `test.out.yaml`:
```bash
#!/bin/bash

filename=test.yaml
out_filename=test.out.yaml

# new items to be added to the list
array=(line4 line5 line6)

# copy filename contents to out_filename.
cp $filename $out_filename

for ((i=0;i<${#array[*]};i++))
do
    {
        dasel put string -f "$out_filename" "list.[]" "${array[$i]}"
    }
done

echo -e "---\n$(cat $out_filename)" > $out_filename
```
E.g.
```bash
$ cat test.yaml
---
list:
 - line1
 - line2
 - line3

$ cat test.out.yaml
---
list:
- line1
- line2
- line3
- line4
- line5
- line6
```

If you'd rather have the list rewritten then you can update `run.sh` as you see fit.

**Reply by TomWright** [reply]
As a side note, your main issue was that you were passing a blank document to dasel for every item within the for loop, rather than using the output from the previous loop.

**Comment by paradajz** [comment]
Exactly the kind of info I was looking for. Thanks a lot!
